### PR TITLE
Add playbooks for HammerDB

### PR DESCRIPTION
**Pull request by mw2q**
These playbooks will set up servers to allow testing with HammerDB.

### CODE REVIEW

File: roles/setup_hammerdb/tasks/configure_postgres_params.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: checkpoint_completion_target
+      value: 0.9
+    - name: max_wal_size
+      value: "30GB"
+    - name: min_wal_size
+      value: "10GB"
+    - name: wal_buffers
+      value: "512MB"
+    - name: jit
+      value: on
```
Comment (jt-edb) [inline roles/setup_hammerdb/tasks/configure_postgres_params.yml:34]: I think we should move this part to `roles/setup_hammerdb/defaults/main.yml`

File: roles/setup_hammerdb/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- name: Include Touchstone installation
+  include_role:
+    name: setup_touchstone
+    tasks_from: main
+
+- name: Create TPROC-C loader.tcl
+  template:
+    dest: loader.tcl
+    src: ./templates/loader.tcl.template
```
Comment (jt-edb) [inline roles/setup_hammerdb/tasks/main.yml:22]: No need to use relative path here, `src: loader.tcl.template` should be enough

File: roles/setup_hammerdb/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    mode: 0644
+  become: no
+
+- name: Create TPROC-C load script
+  template:
+    src: ./templates/load-tproc-c.template
```
Comment (jt-edb) [inline roles/setup_hammerdb/tasks/main.yml:28]: Same comment about relative path usage.

File: roles/setup_hammerdb/tasks/update_pg_hba.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+          'databases': 'all',
+          'contype': 'host',
+          'source': '{{ hostvars[inventory_hostname].hammerdb_server_private_ip | default("127.0.0.1") }}/32',
+          'method': 'trust'
+        },
+        {
+          'users': 'all',
+          'databases': 'all',
+          'contype': 'host',
+          'source': '{{ hostvars['primary1'].private_ip | default("127.0.0.1") }}/32',
```
Comment (jt-edb) [inline roles/setup_hammerdb/tasks/update_pg_hba.yml:21]: For getting primary's private IP address, I would suggest to use the `pg_sr_cluster_nodes` lookup filter:
```yaml
- name: Get the primary information
  set_fact:
    primary_public_ip: "{{ node.ansible_host }}"
    primary_private_ip: "{{ node.private_ip }}"
  loop: "{{ lookup('edb_devops.edb_postgres.pg_sr_cluster_nodes', wantlist=True) }}"
  loop_control:
    loop_var: node
  when:  node.node_type == 'primary'
  no_log: "{{ disable_logging }}"
```

There is no guaranty that primary's inventory hostname is `primary1`.

File: roles/setup_hammerdb/tasks/update_pg_hba.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- name: Initialize the hba_entries variable
+  set_fact:
+    hba_entries: >-
+      [
+        {
+          'users': 'all',
+          'databases': 'all',
+          'contype': 'host',
+          'source': '{{ hostvars[inventory_hostname].hammerdb_server_private_ip | default("127.0.0.1") }}/32',
+          'method': 'trust'
```
Comment (jt-edb) [inline roles/setup_hammerdb/tasks/update_pg_hba.yml:15]: Looks a bit permissive.

File: roles/setup_hammerdbserver/templates/run-tproc-c.template [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+DATE=`date +"%Y-%m-%dT%T%:z"`
+LDIR="${HOME}/results/driver/${DATE}"
+RDIR="${HOME}/results/sut/${DATE}"
+
+ssh -fn {{ hostvars['primary1'].private_ip }} ts-sysstat -o $RDIR
```
Comment (jt-edb) [inline roles/setup_hammerdbserver/templates/run-tproc-c.template:7]: Same comment here about primary's informations.

File: roles/setup_hammerdbserver/templates/run-tproc-c.template [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+(cd ${HOME}/HammerDB-{{ hammerdb_version }} && \\
+ts-sysstat -o $LDIR ./hammerdbcli << !
+source ${HOME}/runner.tcl
+!
+)
+ssh -fn {{ hostvars['primary1'].private_ip }} ts-sysstat -o $RDIR -s
```
Comment (jt-edb) [inline roles/setup_hammerdbserver/templates/run-tproc-c.template:13]: Same comment here about primary's informations.

File: roles/setup_hammerdbserver/templates/runner.tcl.template [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+                if { ![ expr {$x % 60} ] } {
+                        set y [ expr $x / 60 ]
+                }
+                update
+                if {  [ vucomplete ] || $x eq $seconds } { set timerstop 1 }
+        }
+        return
+}
+dbset db pg
+diset connection pg_host {{ hostvars['primary1'].private_ip }}
```
Comment (jt-edb) [inline roles/setup_hammerdbserver/templates/runner.tcl.template:17]: Same comment here about primary's informations.

**Review by jt-edb — COMMENTED** [review]
Just a few minor comments. Thank you Mark!

File: roles/setup_hammerdb/tasks/configure_postgres_params.yml (hunk 1 above)
Comment (vibhorkumar123) [reply roles/setup_hammerdb/tasks/configure_postgres_params.yml:34]: Hi @mw2q, Can you please check if the autotuning role can replace the above setting requirements?

Comment (vibhorkumar123) [reply roles/setup_hammerdb/tasks/configure_postgres_params.yml:34]: Also can you please use FQCN based module call. For example, `template` can be called as `ansible.builtin.template`

File: roles/setup_hammerdb/tasks/update_pg_hba.yml (hunk 5 above)
Comment (mw2q) [reply roles/setup_hammerdb/tasks/update_pg_hba.yml:15]: Yeah, I admit I made that decision for ease of development.  Do we need to use passwords and SSL connections?

Comment (jt-edb) [reply roles/setup_hammerdb/tasks/update_pg_hba.yml:12]: I think we can live with this for the moment.

**Review by jt-edb — APPROVED** [review]
LGTM

**Comment by vibhorkumar123** [comment]
Thank you @jt-edb and @mw2q !
Merging!

**Pull request merged**